### ISSUE TITLE
Pyinstaller Export Not working -wiki tutorial not working either.

### ISSUE THREAD

**Issue by BootsManOut** [TomSchimansky/CustomTkinter#517]
Hello,

I cannot follow the [wiki tutorial](https://github.com/TomSchimansky/CustomTkinter/wiki/Packaging#windows-pyinstaller-auto-py-to-exe) for creating the exe file with pyinstaller.
I try to do it via the console directly.

Already when I run
`pip show customtkinter
`
in the console, I get the message: **_"WARNING: Package(s) not found: customtkinter"_** 

And using the suggested commands with the path to the location of where customtkinter is installed does not work either.
(e.g.: _pyinstaller --noconfirm --onedir --windowed --add-data "C:\Users\Username\OneDrive\Desktop\App Files\customtkinter/customtkinter;customtkinter/"  main.py_ **or** _pyinstaller --noconfirm --onedir --windowed --add-data "C:\Users\Username\OneDrive\Desktop\App Files/customtkinter;customtkinter/"  main.py_)

What can I do to make this work?
I never had issues exporting tkinter apps with other libraries, but I'm not able to make this work in CTkinter, because of the json files.

(btw. I didn't plan to report this, but I might as well mention it here: the **dropdown_fg_color** argument for the CTkCombobox is not recognized, despite being mentioned in the wiki documentation).

**Comment by Akascape** [comment]
@BootsManOut Customtkinter is not packaged automatically because it contains some non-python files. So that's why we have to add its folder in `add-data`.

You may be facing other import errors because of different python versions installed.

I will recommend you to use `Auto-py-to-exe` which is an easy interface for Pyinstaller. Try to package with that.

**Comment by Akascape** [comment]
@BootsManOut I think you have more than one python version installed.
So you have a different version X installed in your environment. (Where Pyinstaller is installed)

But you have installed customtkinter in a different version Y. 

You may be trying to get the module from version X where it is not installed.

**Comment by BootsManOut** [comment]
@Akascape 
You said that customtkinter is not in the same folder as pyinstaller.
Do you mean it's not in the same folder as PIP?

How is the fact that customtkinter is not in the same folder as pyinstaller related to the fact that I cannot run the command
'pip show customtkinter'?

**Comment by Akascape** [comment]
@BootsManOut If you are unable to find the customtkinter installed path then download the repository zip and extract it. Then you will get the customtkinter folder inside it, just copy its path and use it in `-add data`

**Comment by BootsManOut** [comment]
Is this the correct format for it?:

**Comment by Akascape** [comment]
> @Akascape 
> You said that customtkinter is not in the same folder as pyinstaller.
> Do you mean it's not in the same folder as PIP?
> 
> How is the fact that customtkinter is not in the same folder as pyinstaller related to the fact that I cannot run the command
> 'pip show customtkinter'?
> Sorry, I'm a bit confused.

@BootsManOut What I mean to say is that if Pyinstaller is installed in version X, it will collect modules from only version X.

You have customtkinter in version Y, so it will not be able to import it from Y.

**Comment by Akascape** [comment]
> @Akascape 
> yeah that's what I already did.
> 
> 
> Is this the correct format for it?:
> 
> _--add-data "C:\Users\Username\OneDrive\Desktop\App Files\customtkinter/customtkinter;customtkinter/"_
> 
> or does it need to be :
> 
> _--add-data "C:/Users/Username/OneDrive/Desktop/App Files/customtkinter/customtkinter;customtkinter/"_
> 
> or this:
> 
> _--add-data "C:/Users/Username/OneDrive/Desktop/App Files/customtkinter/customtkinter/"_

@BootsManOut I guess third one seems right, but you can get the proper path by clicking the folder properties. Copy that only.

**Comment by BootsManOut** [comment]
> What I mean to say is that if Pyinstaller is installed in version X, it will collect modules from only version X.
> 
> You have customtkinter in version Y, so it will not be able to import it from Y.
> 
> Thats how Pyinstaller works.

Well I was able to export the app together with CTkinter the way I usually export apps, but when I run the app it gives me an error file that it could not find the json files that are used to set the color theme in CTkinter. And that error is called from within the CTkinter script, which means it has succesfully been included by pyinstaller into the exe file.
So it's not a version problem.

**Comment by Akascape** [comment]
@BootsManOut Otherwise use Auto-py-to-exe, you can easily browse and choose the folders/files. Highly recommended.

**Comment by TomSchimansky** [comment]
If you do `python -m pip` you get the correct pip version corresponding to the python version you use.
And to chose a specific python version on Windows in the command line you can use the command `py -<version>`.
So if you use python 3.10 to run your programs you install customtkinter with:
```
py -3.10 -m pip install customtkinter
```
And run your program with:
```
py -3.10 program.py
```
And then this should also work:
```
py -3.10 -m pip show customtkinter 
```

**Comment by TomSchimansky** [comment]
You would install pyinstaller like:
```
py -3.10 -m pip install pyinstaller
```
And use it like this:
```
py -3.10 -m PyInstaller
```
(note the camel case spelling, for whatever reason)

**Comment by BootsManOut** [comment]
> py -3.10 -m pip show customtkinter 
Yes, that's what I needed thanks @TomSchimansky . I got the correct path this way. Installing auto py installer besides not solving the issue, would have just taken away my time.

Now after the json files are included in the exe file, I get another issue with the setuptool module that's used with Ctkinter:

```
  File "main.py", line 4, in <module>
  File "PyInstaller\loader\pyimod02_importers.py", line 499, in exec_module
  File "customtkinter\__init__.py", line 10, in <module>
    from .appearance_mode_tracker import AppearanceModeTracker
  File "PyInstaller\loader\pyimod02_importers.py", line 499, in exec_module
  File "customtkinter\appearance_mode_tracker.py", line 3, in <module>
    from distutils.version import StrictVersion as Version
  File "_distutils_hack\__init__.py", line 80, in create_module
  File "importlib\__init__.py", line 127, in import_module
  File "PyInstaller\loader\pyimod02_importers.py", line 499, in exec_module
  File "setuptools\__init__.py", line 10, in <module>
  File "_distutils_hack\__init__.py", line 80, in create_module
  File "importlib\__init__.py", line 127, in import_module
ModuleNotFoundError: No module named 'setuptools._distutils'
```

Possibly related to [this topic](https://github.com/pypa/setuptools/issues/2353).
However I do not have a **_pyproject.toml_** file, as part of the mentioned solution in that thread, so it seems I cannot make it work.

What can I do in this situation? Do you think you'll ever look into a making the module needing less dependencies for easier portability? I love the look of the UI, but creating the .exe file feels quite difficult.

**Comment by TomSchimansky** [comment]
Maybe you just need pip install pyinstaller?

**Comment by BootsManOut** [comment]
It works now!
So since I had Python 3.9 installed, first I completely re-installed Python as version 3.10.8, and then I updated all the modules of my project, and also updated pip, and then I also ran all the commands you suggested to update pyinstaller etc. And now it works!
Btw. I used this command:
`pyinstaller --noconfirm --onefile --windowed --add-data "C:\Users\USERNAME\AppData\Local\Programs\Python\Python310\Lib\site-packages/customtkinter;customtkinter/" --hidden-import=setuptools --hidden-import=_distutils --icon=images/title.ico  main.py
`
So the "onefile" command actually works without problem, despite being mentioned in the wiki that it shouldn't work.

In any case, thank you very much for your help, @TomSchimansky, it works now!
